src/SapNwRfc/SapTypeMetadata.cs

```diff
@@ -0,0 +1,74 @@
+using System;
+using SapNwRfc.Internal.Interop;
+
+namespace SapNwRfc
+{
+    /// <summary>
+    /// Represents SAP RFC type metadata.
+    /// </summary>
+    public sealed class SapTypeMetadata : ISapTypeMetadata
+    {
+        private readonly RfcInterop _interop;
+        private readonly IntPtr _typeDescription;
+
+        internal SapTypeMetadata(RfcInterop interop, IntPtr typeDescription)
+        {
+            _interop = interop;
+            _typeDescription = typeDescription;
+        }
+
+        /// <inheritdoc cref="ISapTypeMetadata"/>
+        public ISapFieldMetadata GetFieldByIndex(uint index)
```

Comment:
Same remark here, what if we just exposed a `IEnumerable<ISapFieldMetadata> Fields` property?